packages/bruno-app/src/components/Sidebar/Collections/Collection/CollectionItem/GenerateCodeItem/index.js

```diff
@@ -123,11 +133,13 @@ const GenerateCodeItem = ({ collection, item, onClose }) => {
                     item.request.url !== ''
                       ? {
                           ...item.request,
-                          url: interpolatedUrl
+                          url: interpolatedUrl,
+                          headers: headers
```

Comment:
```suggestion
                          url: interpolatedUrl,
                          headers: [...item.request.headers, ...collection.root.request.headers],
```
Merge request headers with collection headers here